contracts/vault/AssetManagement.sol

```diff
@@ -189,24 +189,35 @@ abstract contract AssetManagement is
 
         StrategyData[] memory sortedStrategies = _getSortedStrategiesWithApy(_getStrategiesData());
 
+        uint256 strategyAmountToWithdraw;
         uint256 amountToWithdraw = amount;
 
         for (uint256 i; i < supportedStrategiesVolume; ++i) {
-            try
-                IStrategy(sortedStrategies[i].strategy).withdraw(
-                    sortedStrategies[i].balance <= amountToWithdraw ? sortedStrategies[i].balance : amountToWithdraw
-                )
-            returns (uint256 tryWithdrawnAmount) {
-                amountToWithdraw = tryWithdrawnAmount > amountToWithdraw ? 0 : amountToWithdraw - tryWithdrawnAmount;
-
-                sortedStrategies[i].balance = tryWithdrawnAmount > sortedStrategies[i].balance
-                    ? 0
-                    : sortedStrategies[i].balance - tryWithdrawnAmount;
-            } catch {
-                /// @dev If strategy withdraw fails, try to withdraw from next strategy
+            strategyAmountToWithdraw = sortedStrategies[i].balance <= amountToWithdraw
+                ? sortedStrategies[i].balance
+                : amountToWithdraw;
+
+            if (strategyAmountToWithdraw > 0) {
```

Comment:
I think it will be more readable this way:
```
if (strategyAmountToWithdraw == 0 ) {
    continue;
}

Reply:
I agree.